### ISSUE TITLE
Avoid hardcoding install prefix — always try locating resources relative to the executable

### ISSUE THREAD

**Issue by yggdr** [skyjake/lagrange#395]
Install went fine

```
# cmake -DCMAKE_INSTALL_PREFIX=$HOME/local -P cmake_install.cmake
-- Install configuration: "Release"
-- Installing: /home/yggdrasil/local/bin/lagrange
-- Set runtime path of "/home/yggdrasil/local/bin/lagrange" to "/usr/local/lib"
-- Installing: /home/yggdrasil/local/share/applications/fi.skyjake.Lagrange.desktop
-- Installing: /home/yggdrasil/local/share/icons/hicolor/256x256/apps/fi.skyjake.Lagrange.png
-- Installing: /home/yggdrasil/local/share/metainfo/fi.skyjake.Lagrange.appdata.xml
-- Installing: /home/yggdrasil/local/share/lagrange/resources.lgr
```

Running the installed binary from the build directory, it finds the resources.lgr and everything is fine, but running it outside gives

```
# lagrange
failed to load resources: No such file or directory
```

**Comment by skyjake** [comment]
Which platform are you on, and which version/commit of Lagrange is this?

At least testing now 67f1a5b1 (current dev) on Ubuntu 20.04 it works as it should, finding share/lagrange/resources.lgr in relation to where the executable is located.

**Comment by skyjake** [comment]
Also! Please set `CMAKE_INSTALL_PREFIX` as described in the build instructions: when you run cmake to configure the build, not just when running the install commands.

I just tested setting a different install prefix just for the install commands, and did indeed fail. However, AFAIK that is not the correct way to use CMake's install prefix variable.

**Comment by yggdr** [comment]
I'm on Freebsd13.0/amd64. You are correct, I reread the install instructions again, and indeed the CMAKE_INSTALL_PREFIX is part of the configuration step. That just feels wrong. Building and installing into a specific directory should be orthogonal tasks. Would it be possible to disentangle these parts?
But I'd like to thank you for writing a C-program and making sure it is easily buildable on non-Linux Unixes. That's depressingly rare these days.

**Comment by skyjake** [comment]
> Would it be possible to disentangle these parts?

There is a build option for this already: `ENABLE_RELATIVE_EMBED`. Setting that to `YES` prevents the install prefix from being hardcoded into the executable. In practice, it will then look for the resources in the executable's directory and ../share/lagrange/.

It will not work correctly if the binary directory uses a subdirectory like "x86_64-linux-gnu", but that should not be an issue for you I suppose.

I'll see if I can streamline this a bit more for v1.9.

**Comment by skyjake** [comment]
Changes implemented for v1.9. The hardcoded install prefix, if enabled, is now only one of the paths that get checked.
